engine/hatchery/swarm/swarm.go

```diff
@@ -362,30 +386,43 @@ func (h *HatcherySwarm) createNetwork(name string) error {
 	return err
 }
 
+type containerArgs struct {
+	name, image, network, networkAlias string
+	cmd, env                           []string
+	labels                             map[string]string
+	memory                             int64
+	dockerOpts                         *dockerOpts
+}
+
 //shortcut to create+start(=run) a container
-func (h *HatcherySwarm) createAndStartContainer(name, image, network, networkAlias string, cmd, env []string, labels map[string]string, memory int64) error {
+func (h *HatcherySwarm) createAndStartContainer(cArgs containerArgs) error {
```

Comment:
add this comment: `You could not use this docker options '%s' with a 'shared.infra' hatchery`

Reply:
done

Reply:
here https://github.com/ovh/cds/pull/1481/commits/e8a30ef6304099bac78a8d7bdfafd25e10945b57#diff-5b928cb03058ec28cb0709fcac0a56b6R219